### PR TITLE
[FIX] web_editor: properly sanitize paragraphs in list items

### PR DESCRIPTION
**Pull request by deso-odoo**
Description of the issue this PR addresses:

Paragraphs are not admitted in list items in the editor. But the way they were sanitized away was broken.

After commit [1] contiguous paragraphs in a list item were converted to their own new list item with oe-nested class to remove the bullet point (eg, `<li><p>a</p><p>b</p></li>` became `<li>a</li><li class=oe-nested>b</li>`), which broke some snippet using paragraphs nested within list item.

This commit convert those p tags with attributes to span tags and use `<br>` tags to separate those spans and unwraps those without attributes.

task-3932057

[1]: https://github.com/odoo/odoo/commit/3f9938823189935a013846db9c9a96b5d8bd15d0

### CODE REVIEW

File: addons/web_editor/static/src/js/editor/odoo-editor/src/utils/sanitize.js [hunk 1]
```diff
@@ -248,34 +247,26 @@ class Sanitize {
...
-                } else {
-                    if (classes.length) {
-                        const spanEl = document.createElement('span');
-                        spanEl.setAttribute('class', classes);
-                        spanEl.append(...node.childNodes);
-                        node.replaceWith(spanEl);
-                    } else {
-                        unwrapContents(node);
+                    if (spanEl.style.textAlign) {
+                        spanEl.style.display = 'block';
```
Comment (dmo-odoo) [inline addons/web_editor/static/src/js/editor/odoo-editor/src/utils/sanitize.js:259]: Could you add this comment please?
```js
                        // This is a tradeoff. Ideally, the state of the html
                        // after this function should be reachable by standard
                        // edition means and a span with display block is not.
                        // However, this is required in order to not break the
                        // design of already existing snippets.
```

**Comment by dmo-odoo** [comment]
@robodoo r+

**Comment by deso-odoo** [comment]
@dmo-odoo We are using `replaceWith` here so you might want to override security check.

**Comment by dmo-odoo** [comment]
@robodoo override=ci/security
Reason: The content the node is replaced with is actually its own content wrapped into a span so no new html content is generated apart from the added span, of which we have full control.

**Pull request closed without merging**